messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -231,14 +233,15 @@ private <Q, R> CompletableFuture<QueryResponseMessage<R>> doQuery(
     public <Q, R> Publisher<QueryResponseMessage<R>> streamingQuery(StreamingQueryMessage<Q, R> query) {
         Span span = spanFactory.createChildHandlerSpan(() -> "SimpleQueryBus.streamingQuery", query).start();
         try (SpanScope unused = span.makeCurrent()) {
+            AtomicReference<Throwable> lastError = new AtomicReference<>();
             return Mono.just(intercept(query))
                        .flatMapMany(interceptedQuery -> Mono
                                .just(interceptedQuery)
                                .flatMapMany(this::getStreamingHandlersForMessage)
                                .switchIfEmpty(Flux.error(noHandlerException(interceptedQuery)))
                                .map(handler -> interceptAndInvokeStreaming(interceptedQuery, handler, span))
-                               .skipWhile(ResultMessage::isExceptional)
-                               .switchIfEmpty(Flux.error(noSuitableHandlerException(interceptedQuery)))
+                               .flatMap(new CatchLastError<>(lastError))
+                               .doOnEach(new ErrorIfComplete(lastError, interceptedQuery))
```

Comment:
We need a side effect. I don't see how else to solve it.

Reply:
The side effect is dangerous in the following scenario:

1. the streaming query is invoked
2. it fails, we return the error
3. as a result of an error the user decides to retry by resubscribing to the same publisher
4. we get the same error again (which might not be true since the new execution of the handler could be successful)

I think it would be better to keep `lastError` in the context rather than in the external variable. What do you think?

Reply:
Added the test in a separate commit.